### PR TITLE
BD-3392 Add LINE Login to User ID Reconciliation

### PR DESCRIPTION
**Pull request by rachel-feinberg**
### Summary

Adding a user ID reconciliation method to the docs

### Related PRs, issues, or features (optional)

#BD-3392

### Feature release date (optional)

- N/A

### Contributor checklist

- [ ] I confirm that my PR meets the following:
    - I signed the [Contribution License Agreement (CLA)](https://www.braze.com/docs/cla).
    - My style and voice follows the [Braze Style Guide](https://docs.google.com/document/u/2/d/e/2PACX-1vTluyDFO3ZEV7V6VvhXE4As_hSFwmnFFdU9g6_TrAYTgH1QmbRoEDDdn5GzKAB9vdBbIdyiFdoaJcNk/pub).
    - My content contains correct spelling and grammar.
    - All links are working correctly.
    - If I renamed or moved a file or directory, I set up [URL redirects](https://www.braze.com/docs/contributing/content_management/redirecting_urls) for each file.
    - If I updated or replaced an image, I did not remove the original image file from the repository. (For more information, see [Updating an image](https://www.braze.com/docs/contributing/content_management/images/#updating-an-image)).
    - If my PR is related to a paid SKU, third party, SMS, AI, or privacy, I have received written approval from Braze Legal.

### Submitting for review

If your PR meets the above requirements, select **Ready for review**, then add a reviewer:
- **Non-Braze contributors:** Add `braze-inc/docs-team` as the reviewer.
- **Braze employees:** Have any relevant subject matter experts (like engineers or product managers) review your work first, then add the [tech writer assigned to your specific vertical](https://confluence.braze.com/pages/viewpage.action?pageId=75039896). If you're not sure which tech writer to add, you can add `braze-inc/docs-team` instead. 

Thanks for contributing! We look forward to reading your work.

### CODE REVIEW

File: _docs/_user_guide/message_building_by_channel/line/line_setup.md [hunk 1]
```diff
@@ -274,13 +274,45 @@ LINE IDs are automatically received by Braze when a user follows your channel, o
...
 
-### LINE login
+### LINE Login
 
-This method uses social media logins for reconciliation. When a user logs into your app, they're given the option to use the LINE login to create a user account or log in.
+This method uses social media logins for reconciliation. When a user logs into your app, they're given the option to use [LINE Login](https://developers.line.biz/en/docs/line-login/overview/) to create a user account or log in.
 
-1. Implement the LINE login feature so users can log into your app. For details about implementation, refer to the LINE developer page. After implementation, you can get an access token by calling the [issue access token](https://developers.line.biz/en/reference/line-login/#oauth/) endpoint of the LINE login API. After that, you can get the user’s profile information from ID tokens that are included in the response from the [issue access token](https://developers.line.biz/en/reference/line-login/#oauth/) request.
+{% alert note %}
+To acquire the correct LINE ID for each user, set up LINE Login under the same provider as your Braze integrated LINE official account or channel.
```
Comment (lydia-xie) [inline _docs/_user_guide/message_building_by_channel/line/line_setup.md:282]:
```suggestion
To acquire the correct LINE ID for each user, set up LINE Login under the same provider as your Braze-integrated LINE official account or channel. 
```

File: _docs/_user_guide/message_building_by_channel/line/line_setup.md [hunk 2]
```diff
@@ -274,13 +274,45 @@ LINE IDs are automatically received by Braze when a user follows your channel, o
...
+2. Follow the appropriate steps provided by LINE to implement LINE Login:<br><br>
+  - [Web app directions](https://developers.line.biz/en/docs/line-login/integrate-line-login/)
+  - [Native app directions](https://developers.line.biz/en/docs/line-login/secure-login-process/#using-openid-to-register-new-users)<br><br>Make sure to include `email` in the [scope set up](https://developers.line.biz/en/docs/line-login/integrate-line-login/#scopes) for verification requests. 
+
+{: start="3"}
+3. Use the [Verify ID token call](https://developers.line.biz/en/reference/line-login/#verify-id-token) to acquire the user’s email. 
+
+4. Save the user’s LINE ID (`native_line_id`) to the user’s profile with a matching email in your database, or create a new user profile with the user’s email and LINE ID.
+
+5. Send the new or updated user information to Braze using the [user/track]({{site.baseurl}}/api/endpoints/user_data/post_user_track#track-users/) endpoint, [CSV import]({{site.baseurl}}/user_guide/data_and_analytics/user_data_collection/user_import/#csv-import), or [Cloud Data Ingestion]({{site.baseurl}}/user_guide/data_and_analytics/cloud_ingestion/).
```
Comment (lydia-xie) [inline _docs/_user_guide/message_building_by_channel/line/line_setup.md:296]:
```suggestion
5. Send the new or updated user information to Braze using the [`/user/track` endpoint]({{site.baseurl}}/api/endpoints/user_data/post_user_track#track-users/), [CSV import]({{site.baseurl}}/user_guide/data_and_analytics/user_data_collection/user_import/#csv-import), or [Cloud Data Ingestion]({{site.baseurl}}/user_guide/data_and_analytics/cloud_ingestion/).
```

File: _docs/_user_guide/message_building_by_channel/line/line_setup.md [hunk 3]
```diff
@@ -274,13 +274,45 @@ LINE IDs are automatically received by Braze when a user follows your channel, o
...
+##### New follower uses LINE Login
 
-2. After you get the profile information of a LINE follower, send their LINE user ID to Braze by calling the [user/track]({{site.baseurl}}/api/endpoints/user_data/post_user_track#track-users/) endpoint.
+1. Pre-existing state: No user profile exists in Braze with user’s LINE ID.
+2. The user logs into your app using LINE Login.
+3. LINE provides you the user’s email.
+4. You either:
+  - Update an existing user profile that has that email to also have the user’s LINE ID.
+  - Create a new user profile with the email and LINE ID.
+5. When the user follows your LINE Official Account, Braze receives a follow event and updates the user’s subscription status to “Subscribed”.
```
Comment (lydia-xie) [inline _docs/_user_guide/message_building_by_channel/line/line_setup.md:315]:
```suggestion
1. Pre-existing state: No user profile exists in Braze with the user’s LINE ID.
2. The user logs into your app using LINE Login.
3. LINE provides you with the user’s email.
4. You either:
  - Update an existing user profile with that email to also have the user’s LINE ID.
  - Create a new user profile with the email and LINE ID.
5. When the user follows your LINE Official Account, Braze receives a follow event and updates the user’s subscription status to “Subscribed”.
```

File: _docs/_user_guide/message_building_by_channel/line/line_setup.md [hunk 4]
```diff
@@ -274,13 +274,45 @@ LINE IDs are automatically received by Braze when a user follows your channel, o
...
+
+4. Save the user’s LINE ID (`native_line_id`) to the user’s profile with a matching email in your database, or create a new user profile with the user’s email and LINE ID.
+
+5. Send the new or updated user information to Braze using the [user/track]({{site.baseurl}}/api/endpoints/user_data/post_user_track#track-users/) endpoint, [CSV import]({{site.baseurl}}/user_guide/data_and_analytics/user_data_collection/user_import/#csv-import), or [Cloud Data Ingestion]({{site.baseurl}}/user_guide/data_and_analytics/cloud_ingestion/).
+
+#### Use cases
+
+##### Existing follower uses LINE Login
+
+1. Pre-existing state: An anonymous user was created during initial subscriber sync or after integration through a “follow” event.
```
Comment (lydia-xie) [inline _docs/_user_guide/message_building_by_channel/line/line_setup.md:302]:
```suggestion
1. Pre-existing state: An anonymous user was created during initial subscriber sync or after integration through a “follow” event.
```
What does pre-existing state mean here? 🤔 Like scenario?

**Review by lydia-xie — APPROVED** [review]
looks good! left some suggestions!

File: _docs/_user_guide/message_building_by_channel/line/line_setup.md [hunk 5]
```diff
@@ -274,13 +274,45 @@ LINE IDs are automatically received by Braze when a user follows your channel, o
...
+##### New follower uses LINE Login
 
-2. After you get the profile information of a LINE follower, send their LINE user ID to Braze by calling the [user/track]({{site.baseurl}}/api/endpoints/user_data/post_user_track#track-users/) endpoint.
+1. Pre-existing state: No user profile exists in Braze with the user’s LINE ID.
+2. The user logs into your app using LINE Login.
+3. LINE provides you with the user’s email.
+4. You either:
+  - Update an existing user profile with that email to also have the user’s LINE ID.
+  - Create a new user profile with the email and LINE ID.
+5. When the user follows your LINE Official Account, Braze receives a follow event and updates the user’s subscription status to “Subscribed”.
```
Comment (rachel-feinberg) [inline _docs/_user_guide/message_building_by_channel/line/line_setup.md:315]:
```suggestion
**Scenario:** No user profile exists in Braze with the user’s LINE ID.

1. The user logs into your app using LINE Login.
2. LINE provides you with the user’s email.
3. You either:
  - Update an existing user profile with that email to also have the user’s LINE ID.
  - Create a new user profile with the email and LINE ID.
4. When the user follows your LINE Official Account, Braze receives a follow event and updates the user’s subscription status to “Subscribed”.
```

File: _docs/_user_guide/message_building_by_channel/line/line_setup.md [hunk 6]
```diff
@@ -274,13 +274,45 @@ LINE IDs are automatically received by Braze when a user follows your channel, o
...
+5. Send the new or updated user information to Braze using the [`/user/track` endpoint]({{site.baseurl}}/api/endpoints/user_data/post_user_track#track-users/), [CSV import]({{site.baseurl}}/user_guide/data_and_analytics/user_data_collection/user_import/#csv-import), or [Cloud Data Ingestion]({{site.baseurl}}/user_guide/data_and_analytics/cloud_ingestion/).
+
+#### Use cases
+
+##### Existing follower uses LINE Login
+
+1. Pre-existing state: An anonymous user was created during initial subscriber sync or after integration through a “follow” event.
+2. The user logs into your app using LINE Login.
+3. LINE provides you the user’s email.
+4. You send Braze the updated user (the existing user profile with that email to add the LINE ID) or you update the anonymous user with the email.
```
Comment (rachel-feinberg) [inline _docs/_user_guide/message_building_by_channel/line/line_setup.md:305]:
```suggestion
**Scenario:** An anonymous user was created during initial subscriber sync or after integration through a “follow” event.

1. The user logs into your app using LINE Login.
2. LINE provides you the user’s email.
3. You send Braze the updated user (the existing user profile with that email to add the LINE ID) or you update the anonymous user with the email.
```

**Pull request merged**